### PR TITLE
[Streaming] Add _event_string unit tests for `StreamCompletion` event

### PR DESCRIPTION
**Pull request by royischoss**
Adding unit tests for _event_string with `StreamCompletion`

### CODE REVIEW

File: tests/test_streaming.py [hunk 1]
```diff
@@ -1892,3 +1892,131 @@ def select_outlets(self, event):
...
+        self.logs.append(("warn", args, kwargs))
+
+    def info(self, *args, **kwargs):
+        self.logs.append(("info", args, kwargs))
+
+    def debug(self, *args, **kwargs):
+        self.logs.append(("debug", args, kwargs))
+
+
+class MockContext:
```
Comment (gtopper) [inline tests/test_streaming.py]: We already have these in `test_flow.py`, though. Can extract and use to avoid duplication?

File: tests/test_streaming.py [hunk 2]
```diff
@@ -1892,3 +1892,131 @@ def select_outlets(self, event):
...
+        # The logs should contain references to the step names showing flow progression
+        assert "StreamingMap" in all_log_messages or "Collector" in all_log_messages
+
+    def test_event_string_with_stream_completion_no_original_event(self):
+        """Test that _event_string handles StreamCompletion with None original_event.
+
+        Edge case: StreamCompletion.body returns None when original_event is None.
+        The _event_string method should handle this gracefully.
+        """
+        from storey.flow import Flow
```
Comment (gtopper) [inline tests/test_streaming.py]: Imports should be at the top unless there's a good reason to put them in a local scope, which isn't the case here. Applies elsewhere as well.

File: tests/test_streaming.py [hunk 3]
```diff
@@ -1892,3 +1892,131 @@ def select_outlets(self, event):
...
+
+        completion = StreamCompletion("streaming_step", None)  # type: ignore[arg-type]
```
Comment (gtopper) [inline tests/test_streaming.py]: Should probably not test it with `None` if it's not a valid parameter.

File: tests/test_streaming.py [hunk 4]
```diff
@@ -1892,3 +1892,131 @@ def select_outlets(self, event):
...
+
+        # _event_string should handle StreamCompletion with None original_event without error
+        result = Flow._event_string(completion)
+
+        # Should produce a valid string with body=None
+        assert isinstance(result, str)
+        assert "body=None" in result
+
+    def test_event_string_with_stream_completion_with_error(self):
```
Comment (gtopper) [inline tests/test_streaming.py]: Seems like a redundant test, given that the error doesn't seem relevant here, and we already test the same thing in `test_event_string_with_stream_completion`?

File: storey/dtypes.py [hunk 5]
```diff
@@ -160,6 +160,10 @@ def __repr__(self):
     def id(self):
         return self.original_event.id if self.original_event else None
 
+    @property
+    def body(self):
+        return self.original_event.body if self.original_event else None
```
Comment (gtopper) [inline storey/dtypes.py]: This one actually doesn't seem to make much sense... Why should we want `StreamCompletion` to have the original event's body?

**Comment by gtopper** [comment]
Oh, also please fix the title :)

File: storey/dtypes.py (hunk 5 above)
Comment (royischoss) [reply storey/dtypes.py]: we need to have a body or change the _event_string

File: tests/test_streaming.py [hunk 6]
```diff
@@ -36,6 +37,7 @@
 )
 from storey.dtypes import Event, StreamChunk, StreamCompletion
 from storey.flow import _is_generator
+from tests.test_flow import MockContext, MockLogger
```
Comment (gtopper) [inline tests/test_streaming.py]: I think we should avoid importing the massive test_flow, and should instead extract these classes to a helper file.

File: storey/flow.py [hunk 7]
```diff
@@ -320,7 +320,7 @@ def _event_string(event):
             result += f"processing_time={event.processing_time}, "
         if getattr(event, "path", None):
             result += f"path={event.path}, "
-        result += f"body={event.body})"
+        result += f"body={event.body})" if hasattr(event, "body") else ""
```
Comment (gtopper) [inline storey/flow.py]: The paren needs to be present either way, though. How about:
```suggestion
    if hasattr(event, "body"):
        result += f"body={event.body}" 
    result += ")"
```

**Review by gtopper — COMMENTED** [review]
Small bug, minor test housekeeping. LGTM otherwise  🚀

File: storey/flow.py [hunk 8]
```diff
@@ -320,7 +320,8 @@ def _event_string(event):
             result += f"processing_time={event.processing_time}, "
         if getattr(event, "path", None):
             result += f"path={event.path}, "
-        result += f"body={event.body})"
+        if hasattr(event, "body"):
+            result += f"body={event.body})"
```
Comment (gtopper) [inline storey/flow.py]: The paren needs to be taken out of the conditional as per my earlier code suggestion.

**Review by gtopper — APPROVED** [review]
LGTM ✅
